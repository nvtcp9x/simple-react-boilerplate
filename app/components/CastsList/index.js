// @flow

import * as React from 'react';
import styled from 'styled-components';
import { Card, CardBody, CardText } from 'reactstrap'

import { URL_IMG, IMG_SIZE_SMALL } from '../../../utils/constants'

const Wrapper = styled.div`
  display: flex;
  white-space: nowrap;
  overflow: auto;
`

const StyledCard = styled(Card)`
  min-width: auto;
  border: 1px solid #e3e3e3;
  margin-right: 10px;
  margin-bottom: 7px;
  box-shadow: 0 2px 8px rgba(0,0,0,0.1);
`
const StyledImg = styled.img`
  max-height: 150px;
  margin: 0 auto;
  display: block;
`
const CastsList = ({ casts }: Array) =>
  (
    <Wrapper>
      { casts.map(cast =>
        (
          <StyledCard key={cast.id}>
            <StyledImg
              top
              src={cast.profile_path ? `${URL_IMG}${IMG_SIZE_SMALL}${cast.profile_path}` : '/images/face.jpg'}
              alt={cast.character}
              className="img-fluid"
            />
            <CardBody>
              <CardText>{cast.name}</CardText>
              <CardText>{cast.character}</CardText>
            </CardBody>
          </StyledCard>
        ))}
    </Wrapper>
  )

export default CastsList

