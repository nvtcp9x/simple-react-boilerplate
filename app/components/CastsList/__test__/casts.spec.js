import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import CastsList from '../index';

Enzyme.configure({ adapter: new Adapter() });

describe('CastsList component', () => {
  const mockProps = {
    casts: [{
      id: 1,
      profile_path: 'hfdkjfd',
      character: 'fdo',
      name: 'fdjfkd',
    },
    ],
  };

  const component = shallow(<CastsList {...mockProps} />);

  it('should ...', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
