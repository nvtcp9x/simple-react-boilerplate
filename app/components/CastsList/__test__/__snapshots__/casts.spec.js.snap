// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CastsList component should ... 1`] = `
<styled.div>
  <div
    className="card"
    key="1"
  >
    <img
      alt="fdo"
      className="card-img-top"
      src="https://image.tmdb.org/t/p/w138_and_h175_facehfdkjfd"
    />
    <div
      className="card-body"
    >
      <p
        className="card-text"
      >
        fdjfkd
      </p>
      <p
        className="card-text"
      >
        fdo
      </p>
    </div>
  </div>
</styled.div>
`;
